granulate_utils/metadata/databricks_client.py

```diff
@@ -44,45 +50,81 @@ def get_webui_address() -> Optional[str]:
     def get_job_name(self) -> Optional[str]:
         # Retry in case of a connection error, as the metrics server might not be up yet.
         for i in range(MAX_RETRIES):
-            time.sleep(10)
             try:
-                return self._get_job_name_impl()
-            except Exception:
-                self.logger.exception("Got Exception while collecting Databricks job name.")
+                if cluster_metadata := self._cluster_all_tags_metadata():
+                    # Got the job name, no need to retry.
+                    name = self._get_name_from_metadata(cluster_metadata)
+                    if name:
+                        self.logger.debug("Found name in metadata.", job_name=name, cluster_metadata=cluster_metadata)
+                        return name
+                    else:
+                        self.logger.debug("Failed to extract name from metadata.", cluster_metadata=cluster_metadata)
+                        return None
+                else:
+                    # No job name yet, retry.
+                    time.sleep(15)
```

Comment:
Constify 15 at the top of the file

Reply:
Also, 15 is very high. This blocks the profiler initialization. Why so high?

Reply:
please note I've changed to `while` instead of `for`
and now the retry interval is 1 second for 5 minutes
9b60264